### DISCUSSION TITLE
Reasoning about Color Scheme

### DISCUSSION THREAD

**Discussion by lioneltay** [chakra-ui/chakra-ui#5938] [General]
The `colorScheme` prop uses a set of colors ranging between 50-900 which is used differently by different components. I was wondering if there is a way to "reason" about what colors should be used when. 

EG (for button, and most components) it seems that 500 is the standard "base" color for light mode, and 200 is the "base" for dark, (at least for colors, not black/white/grey). Then in a different state, it is common to go up or down 100, etc.

Is there a general guide/methodology when using a particular color lightness? I wasn't able to find one in the docs, although I may have missed it.

There are also a few recommended color scheme generators such as [SmartSwatch](https://smart-swatch.netlify.app/#C70833) and [Themera](https://themera.vercel.app/). But it seems that the color you use to generate the pallete is not actually in the pallete itself. Is there a way to generate a pallete where you set a particular color, eg red.400 to a specific color, and have the other colors generated based on that? It seems difficult to match a predefined design without being able to fix a particular color. Or maybe I am going about this wrong.

**Comment by BillyPurvis** [comment]
Not a full answer, but we have this exact issue. For the most part, we tend to think of the 500 as the middle-ground. It provides a guarantee that below 500 it is darker by roughly `x` percentage, and above, the same, but darker. It's also very predictable if each colour works like this so a component can apply any colour and the behaviour would be the same (ergo the button gets darker when you hover on it, regardless). 

Though, Chakra does follow a lot of Tailwind's ideas and practices for styling and Tailwind's docs does answer your question in some regard: https://tailwindcss.com/docs/customizing-colors#generating-colors

As for being used as the 'base', I'd assume this is for consistency in branding. You'd likely want all of your components, especially interactive ones, to be the same colour as a default. So your Buttons, Icon buttons, checkboxes and so forth are all the same colour when using the same colour scheme. 

Regarding the generators, we found this issue and we're likely to just generate our own swatch using our current colour value as the 'base' weight of 500. As long as each colour swatch we generate follows the same increase/decreases for each weight, it should be okay, but we might make slight adjustments based on accessibility colour contrasts and generally what works for that color, as long as we don't randomly make colour A's 400th weight much much darker than colour B's 400th weight in comparison to it's 500th, otherwise you lose predictability across your colours.
